File: fingerprint-ui-edu/src/main/java/com/fingerprint/service/institution/InstitutionService.java
package com.fingerprint.service.institution;

import com.fingerprint.model.Institution;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

/**
 * Created by lauearo on 25/05/2017.
 */
@FeignClient("institution")
public interface InstitutionService {

    @RequestMapping(method = RequestMethod.GET, value = "/api/{id}",
            produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
    Institution get(@PathVariable("id") long id);

    @RequestMapping(method = RequestMethod.PUT, value = "/api/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
    Institution update(@PathVariable("id") long id, Institution institution);
}

File: fingerprint-ui-edu/src/main/resources/static/webapp/js/services/institutionService.js
/**
 * Created by lauearo on 20/06/2017.
 */
define([], function () {
    function institutionService($resource) {
        var institution = $resource('/ins/:insId', {
            insId: '@id'
        });

        return institution;
    }

    institutionService.$inject = ['$resource'];
    return institutionService;
});